### DISCUSSION TITLE
How to get a file from XmlDocument ?

### DISCUSSION THREAD

**Discussion by ben55j** [renggli/dart-xml#159] [Q&A] [answered]
Hello, 

I am looking to upload to Firebase Storage a XmlDocument that I have created in my Flutter app.
The method for uploading requires a File as parameter, and I don't know how to get it from the XmlDocument.

Could you help please ?
Thank you, 

Benjamin

**Comment by renggli** [comment, marked as answer]
See the documentation on [how to read and write](https://docs.flutter.dev/cookbook/persistence/reading-writing-files) files, i.e. 

```dart 
final file = new File('filename.txt');
final stream = await file.writeAsString(document.toString());
```
